### DISCUSSION TITLE
Road to first class routes in lagoon api

### DISCUSSION THREAD

**Discussion by Schnitzel** [amazeeio/lagoon#3467] [Roadmap]
1. create new lagoon environment variable which expects a json object the same structure as the `routes` key in the `.lagoon.yml` which is passed to lagoon-build-deploy pod and then merged with what is found in the `.lagoon.yml` for the environment. This is a bit risky but the easiest to implement.
2. implement route objects in lagoon api and pass them to lagoon-build-deploy pod which then merges them
3. Have lagoon-build-deploy fill route objects in lagoon api after a deployment and deprecate `.lagoon.yml` routes setting
4. with new Lagoon Major Vesrion: Only use route objects in lagoon api and no longer deploy `.lagoon.yml` routes

**Comment by lind101** [comment]
Hey all,

We would love to see this implimented! Let me know if you need any help testing!

**Comment by shreddedbacon** [comment]
We've discussed a proof of concept for getting this in by using an environment variable defined in the API that contains the data structure that would be passed to the build process to create them at the next deployment.

The idea will be that we will merge this into the existing `.lagoon.yml` some how to make sure that the existing process of creating routes through a build remain the same.

I'm working on if this is possible to do, as the `.lagoon.yml` structure isn't really ideal.
